solidity/test/random_beacon_operator/TestRelayEntryTimeout.js

```diff
@@ -41,61 +88,71 @@ describe("KeepRandomBeaconOperator/RelayEntryTimeout", function() {
     await restoreSnapshot()
   });
 
-  it("should not throw an error when entry is in progress and " +
-     "block number > relay entry timeout", async () => {
-    await operatorContract.sign(
-      requestCounter, blsData.previousEntry, {value: fee, from: accounts[0]}
-    );
-
-    await time.advanceBlockTo(blocksForward.addn(await web3.eth.getBlockNumber()))
-
-    await operatorContract.sign(
-      requestCounter, blsData.previousEntry, {value: fee, from: accounts[0]}
-    );
-
-    assert.equal(
-      (await operatorContract.getPastEvents())[0].event, 
-      "RelayEntryRequested", 
-      "RelayEntryRequested event should occur on operator contract"
-    );
-  });
-
-  it("should throw an error when entry is in progress and " + 
-     "block number <= relay entry timeout", async () => {
-    await operatorContract.sign(
-      requestCounter, blsData.previousEntry, {value: fee, from: accounts[0]}
-    );
-
-    await expectRevert(
-      operatorContract.sign(requestCounter, blsData.previousEntry, {value: fee, from: accounts[0]}), 
-      "Beacon is busy"
-    );
-  });
-
-  it("should not throw an error when entry is not in progress and " + 
-     "block number > relay entry timeout", async () => {
-    await operatorContract.sign(
-      requestCounter, blsData.previousEntry, {value: fee, from: accounts[0]}
+  describe("request for a new relay entry", async () => {
+    it("should be accepted when no other request is in progress", async () => {
+      const receipt = await requestRelayEntry()
+      await expectEvent(receipt, "RelayEntryRequested")
+    })
+
+    it("should be rejected when another request is in progress", async () => {
+      await requestRelayEntry()
+      await expectRevert(requestRelayEntry(), "Beacon is busy")
+    })
+
+    it("should be rejected when another request is in progress even if it timed out", async () => {
+      const timeout = await operatorContract.relayEntryTimeout()      
+      await requestRelayEntry()
+      await time.advanceBlockTo((await time.latestBlock()).add(timeout))    
+      await expectRevert(requestRelayEntry(), "Beacon is busy")
+    })
+
+    it("should be retried when another request timed out and it's been reported", async () => {
+      const timeout = await operatorContract.relayEntryTimeout()      
+      await requestRelayEntry()
+      await time.advanceBlockTo((await time.latestBlock()).add(timeout))
+      const receipt = await operatorContract.reportRelayEntryTimeout({from: thirdParty})
+      
+      expect(await operatorContract.isEntryInProgress()).to.be.true;
+      await expectEvent(receipt, "RelayEntryRequested") 
```

Comment:
Looks like it may be worth explicitly adding an assertion that `requestRelayEntry` gives a `Beacon is busy` here, perhaps?

Reply:
Sure, it doesn't hurt!